lib/TwoFactorAuth.php

```diff
@@ -157,29 +158,23 @@ public function getQRText(string $label, string $secret): string
         return 'otpauth://totp/' . rawurlencode($label)
             . '?secret=' . rawurlencode($secret)
             . '&issuer=' . rawurlencode((string)$this->issuer)
-            . '&period=' . intval($this->period)
+            . '&period=' . $this->period
             . '&algorithm=' . rawurlencode(strtoupper($this->algorithm->value))
-            . '&digits=' . intval($this->digits);
+            . '&digits=' . $this->digits;
     }
 
-    /**
-     * @throws TwoFactorAuthException
-     */
     public function getQrCodeProvider(): IQRCodeProvider
     {
         // Set default QR Code provider if none was specified
-        if (null === $this->qrcodeprovider) {
-            return $this->qrcodeprovider = new QRServerProvider();
-        }
-        return $this->qrcodeprovider;
+        return $this->qrcodeprovider ?? ($this->qrcodeprovider = new QRServerProvider());
```

Comment:
isn't this what `??=` is for? (Null Coalescing Assignment)

Reply:
Good call. Fixed in https://github.com/RobThree/TwoFactorAuth/pull/108/commits/9bc454f42501a89ce98860e1e9b7b774e555bf38.